cassdegrees/templates/managecourses.html

```diff
@@ -1,148 +1,76 @@
-                            </p>
-
-                            <p>
-                                <label for="id">Course Code {{ required }}</label>
-                                <input class="text tfull" id="text" type="text" name="code" minlength="8" maxlength="8" size="8" aria-required="true" required>
-                            </p>
-
-                            <p>
-                                <label for="text">Year {{ required }}</label>
-                                <input class="text tfull" id="number" type="number" name="year" min="1990" aria-required="true" required>
-                            </p>
 
-                            <p>
-                                <label for="text">Semester(s)</label>
-                                <select class="text tfull" name="semesters[]" size="2" multiple>
-                                    <option value="semester1">Semester 1</option>
-                                    <option value="semester2">Semester 2</option>
-                                </select>
-                            </p>
-
-                             <p>
-                                <label for="text">Units {{ required }}</label>
-                                 <!-- max value is chosen arbitrarily -->
-                                <input class="text tfull" id="number" type="number" name="units" min="0" step="6" aria-required="true" required>
-                            </p>
-
-                            <p>
-                                <label for="text">Requisite Course(s)</label>
-                                <select class="text tfull" name="requisites[]" size="5" multiple>
-                                    {% for course in courses %}
-                                        <option value="{{ course.code }}">{{ course.code }}</option>
-                                    {% endfor %}
-                                </select>
-                            </p>
-
-                            <p>
-                                <label for="text">Incompatible Course(s)</label>
-                                <select class="text tfull" name="incompatibles[]" size="5" multiple>
-                                    {% for course in courses %}
-                                        <option value="{{ course.code }}">{{ course.code }}</option>
-                                    {% endfor %}
-                                </select>
-                            </p>
-                        </fieldset>
-                    {% endwith %}
-
-                    <p class="text-right">
-                        <input class="btn-uni-grad btn-large" type ="submit" value="{{ action }}">
+    <div id="{{ action }}Page" {% if action == 'Delete' or render.hide_form %}hidden{% endif %}>
+        <form class="anuform" action="/manage_courses/?action={{ action }}" method="post">
```

Comment:
We can't do this action stuff through POST?

Reply:
Which line are you talking about? I don't understand what you are suggesting.

Reply:
` action="/manage_courses/?action={{ action }}"`
Why are we mixing GET and POST?

Reply:
How are we mixing GET and POST? The form most certainly states that the method is post. Adding "?action=" is not an indicator that it is a GET request.

Also, while we can create a new hidden input with the action variable from managecourses.html, there is this one problem. From list.html, there is no way to determine which button the user will press on until they do and that 'submits' the form immediately. Therefore, there is no possible way to quickly change this hypothetical hidden input value to the one we want ("Add", "Edit" or "Delete") before the form is submitted. Hence, we will have to keep it this way.

Reply:
Action in a URL is a [GET parameter](https://en.wikipedia.org/wiki/Query_string). Adding action in the URL there is by definition a query string, which is what is used for GET requests. This just seems a little weird to me.

Its not too big on an issue, but something to note.

Reply:
Actually, according to HTTP specifications, it is valid to have a query string for POST requests:
https://stackoverflow.com/questions/14710061/is-it-valid-to-combine-a-form-post-with-a-query-string/14710450
https://www.w3.org/Protocols/rfc2616/rfc2616.html

But I did just find out that it is generally not a good idea, so you were right to point that line out. Unfortunately, we still have to leave it this way due to the button problem :(

Reply:
Na, totally fair enough - just as long as this is noted, I personally have no issue (as you said, it isn't technically incorrect).